src/Sylius/Component/Core/spec/Model/ChannelSpec.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Core\Model;
+
+use Doctrine\Common\Collections\Collection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Addressing\Model\ZoneInterface;
+use Sylius\Component\Channel\Model\Channel as BaseChannel;
+use Sylius\Component\Core\Model\Channel;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Currency\Model\CurrencyInterface;
+use Sylius\Component\Locale\Model\LocaleInterface;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+final class ChannelSpec extends ObjectBehavior
+{
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(Channel::class);
+    }
+
+    function it_implements_a_channel_interface()
+    {
+        $this->shouldImplement(ChannelInterface::class);
+    }
+
+    function it_extends_an_channel()
+    {
+        $this->shouldHaveType(BaseChannel::class);
+    }
+
+    function it_has_no_base_currency_by_default()
+    {
+        $this->getBaseCurrency()->shouldReturn(null);
+    }
+
+    function its_base_currency_is_mutable(CurrencyInterface $baseCurrency)
+    {
+        $this->setBaseCurrency($baseCurrency);
+        $this->getBaseCurrency()->shouldReturn($baseCurrency);
+    }
+
+    function it_has_no_default_locale_by_default()
+    {
+        $this->getDefaultLocale()->shouldReturn(null);
+    }
+
+    function its_default_locale_is_mutable(LocaleInterface $defaultLocale)
+    {
+        $this->setDefaultLocale($defaultLocale);
+        $this->getDefaultLocale()->shouldReturn($defaultLocale);
+    }
+
+    function it_has_no_default_tax_zone_by_default()
+    {
+        $this->getDefaultTaxZone()->shouldReturn(null);
+    }
+
+    function its_default_tax_zone_is_mutable(ZoneInterface $defaultTaxZone)
+    {
+        $this->setDefaultTaxZone($defaultTaxZone);
+        $this->getDefaultTaxZone()->shouldReturn($defaultTaxZone);
+    }
+
+    function it_has_no_tax_calculation_strategy_by_default()
+    {
+        $this->getTaxCalculationStrategy()->shouldReturn(null);
+    }
+
+    function its_tax_calculation_strategy_is_mutable($taxCalculationStrategy)
+    {
+        $this->setTaxCalculationStrategy($taxCalculationStrategy);
+        $this->getTaxCalculationStrategy()->shouldReturn($taxCalculationStrategy);
+    }
+
+    function it_initializes_currencies_collection_by_default()
+    {
+        $this->getCurrencies()->shouldHaveType(Collection::class);
+    }
+
+    function it_adds_a_currency(CurrencyInterface $currency)
```

Comment:
It definitely does not.
You got the subject wrong, you don't use the channel to add currencies, you add currencies to the channel.
So something more like `it_can_have_a_currency_added/removed` or just `its_currencies_collection_is_mutable` and put both adding and removing in one function.
Same thing with locales.